Implement quit-applications

mackopes opened the pull request:
## Description

<!-- A summary of your change. If you add a new extension or command, explain what it does. -->
A simple extension to see all running applications in one place and quit the selected one.

I am aware that a similar functionality exists in the base Raycast app. This extension, however, offers additional functionality:
- The window does not close after you quit an app, so it is suitable for quitting multiple apps at once.
- It is easy to see all open and running apps at the same time to easily decide what to quit. It is much harder to miss an app this way.
- For users migrating from Alfred (like I did) with already built muscle memory, this extension offers a familiar experience.

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

<img width="1000" alt="screen1" src="https://user-images.githubusercontent.com/25415255/181655443-f5d3e7fe-354f-4896-bdfe-f798803b508f.png">


## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

danielsequeira commented on extensions/quit-application/package.json:
```diff
@@ -0,0 +1,42 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "quit-application",
+  "title": "Quit Application",
```
I would suggest rename it to plural, since it actually allows you to quite multiple applications within the same command
```suggestion
  "title": "Quit Applications",
```

danielsequeira commented on extensions/quit-application/package.json:
```diff
@@ -0,0 +1,42 @@
+    {
+      "name": "index",
+      "title": "Quit Application",
+      "subtitle": "System",
```
Could you use another subtitle or remove it completely? It would be better to now mix up public extensions with our Raycast native System commands

danielsequeira commented on extensions/quit-application/package.json:
```diff
@@ -0,0 +1,42 @@
+  "commands": [
+    {
+      "name": "index",
+      "title": "Quit Application",
```
```suggestion
      "title": "Quit Applications",
```

danielsequeira commented on extensions/quit-application/src/index.tsx:
```diff
@@ -0,0 +1,61 @@
+                <Action
+                  title="Quit application"
+                  onAction={() => {
+                    // closeMainWindow({ clearRootSearch: true });
```
remove commented code?

danielsequeira commented:
Hi @mackopes!

Thanks for your contribution 🙌 

I just added a few comments, nothing major, just nitpicking about naming.

Besides that I noticed that your extension icon is not 512x512 actually, it has a lot of margin, which makes it not look so good in Raycast, could you fix it?

Also when entering the command, for a split second we are presented with "No Results" screen (check screenshot below). I think you could fix this by adding some loading rule to your list.
![Raycast 2022-08-03 at 16 17 42](https://user-images.githubusercontent.com/12091319/182631841-80d58bb5-eec2-44f0-975f-ed81163f2d4c.png)

danielsequeira commented:
@mackopes ping here, do you still plan to finalise this PR?

mackopes commented:
@danielsequeira Apologies for the delay and thanks for checking-in! I still plan to finalise the PR. I will do so by the end of the week and tag you again :)

mackopes commented:
@danielsequeira
Thanks a lot for such a good feedback! I am not very familiar with type script, so any advice was very welcome!

I addressed all your suggestions - Using plural in Quit Applications, removing comment, making the figure larger and adding the loading rule.

Could you please review this again whenever you have some time? Lmk if there's anything else I can do to make it better.

danielsequeira commented on extensions/quit-application/CHANGELOG.md:
```diff
@@ -0,0 +1,3 @@
+# Quit Application Changelog
```
```suggestion
# Quit Applications Changelog
```

danielsequeira commented on extensions/quit-application/README.md:
```diff
@@ -0,0 +1,3 @@
+# Quit application
```
```suggestion
# Quit Applications
```

danielsequeira commented on extensions/quit-application/package.json:
```diff
@@ -0,0 +1,41 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "quit-application",
```
```suggestion
  "name": "quit-applications",
```

danielsequeira commented on extensions/quit-application/package.json:
```diff
@@ -0,0 +1,41 @@
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "quit-application",
+  "title": "Quit Applications",
+  "description": "Quit apps directly from Raycast",
```
```suggestion
  "description": "Quit applications directly from Raycast",
```

danielsequeira commented on extensions/quit-application/package.json:
```diff
@@ -0,0 +1,41 @@
+    {
+      "name": "index",
+      "title": "Quit Applications",
+      "description": "Quit apps directly from Raycast",
```
```suggestion
      "description": "Quit applications directly from Raycast",
```

danielsequeira commented on extensions/quit-application/src/index.tsx:
```diff
@@ -0,0 +1,62 @@
+            actions={
+              <ActionPanel>
+                <Action
+                  title="Quit application"
```
```suggestion
                  title="Quit"
```

danielsequeira commented on extensions/quit-application/CHANGELOG.md:
```diff
@@ -0,0 +1,3 @@
+# Quit Application Changelog
+
+## [Initial Quit Application implementation] - 2022-07-29
```
```suggestion
## [Initial Quit Applications implementation] - 2022-08-16
```

danielsequeira commented:
Thanks for the changes @mackopes. I just left a few more suggestions to keep consistency in terms of naming. After these are applied I think we can merge it.

mackopes commented:
Thanks @danielsequeira for the good review, could you give it 👀 one more time?
I updated all quit application references to plural (as well as the directory name to stay consistent).

danielsequeira commented:
Thanks @mackopes.

Sorry for the nitpicking but the extension icon doesn't look great in Raycast. The resolution looks a bit off.
![Screenshot 2022-08-22 at 15 06 35](https://user-images.githubusercontent.com/12091319/185928606-38753a60-b115-45a6-934e-7db2b7476d13.png)
![Screenshot 2022-08-22 at 15 06 27](https://user-images.githubusercontent.com/12091319/185928619-4b776863-3558-48bd-8860-cf43db836bf8.png)

Would it be possible for you to find a better one?

danielsequeira commented:
@mackopes are you still planning to pick this up? 🙂

danielsequeira commented:
@mackopes I can also help you create an icon if this would help

mackopes commented:
Hi @danielsequeira, 

apologies for the delay. I came up with a new design for the icon. I believe maybe more minimalistic design will look better. What do you think about this one? I was thinking about having different colour schemes for dark and light system appearance, but this one seems to work well for both.

<img width="773" alt="image" src="https://user-images.githubusercontent.com/25415255/198273888-17ad15a3-ad39-41e6-992e-03c19567bdfa.png">

<img width="748" alt="Screenshot 2022-10-27 at 12 34 51" src="https://user-images.githubusercontent.com/25415255/198274058-1ce03a9c-5fee-4167-91bf-24549966585b.png">

danielsequeira commented:
@mackopes the new icon looks good, thanks for updating it 🙂 the last thing you are missing though is updating the screenshots in the metadata folder, since they still display the old one.

mackopes commented:
@danielsequeira  updated now! :)
